samples/android_main.cc

```diff
@@ -23,23 +23,34 @@ extern int main(int argc, const char** argv);
 extern "C" JNIEXPORT JNICALL int Java_com_google_amber_Amber_androidMain(
     JNIEnv* env,
     jobject,
-    jstring args_jstring) {
+    jobjectArray args,
+    jstring stdoutFile,
+    jstring stderrFile) {
+  const char* stdout_file_cstr = env->GetStringUTFChars(stdoutFile, NULL);
+  const char* stderr_file_cstr = env->GetStringUTFChars(stderrFile, NULL);
+
   // Redirect std output to a file
-  freopen("/sdcard/amberlog.txt", "w", stdout);
-  freopen("/sdcard/amberlog.txt", "a", stderr);
+  freopen(stdout_file_cstr, "w", stdout);
+  freopen(stderr_file_cstr, "w", stderr);
+
+  env->ReleaseStringUTFChars(stdoutFile, stdout_file_cstr);
+  env->ReleaseStringUTFChars(stderrFile, stderr_file_cstr);
+
+  int arg_count = env->GetArrayLength(args);
 
-  std::string args(env->GetStringUTFChars(args_jstring, NULL));
+  std::vector<std::string> argv_string;
+  for (int i = 0; i < arg_count; i++) {
+    jstring js = static_cast<jstring>(env->GetObjectArrayElement(args, i));
+    const char* arg_cstr = env->GetStringUTFChars(js, NULL);
+    argv_string.push_back(arg_cstr);
+    env->ReleaseStringUTFChars(js, arg_cstr);
+  }
 
-  // Parse argument string and add -d by default
-  std::stringstream ss(args);
-  std::vector<std::string> argv_string{std::istream_iterator<std::string>{ss},
-                                       std::istream_iterator<std::string>{}};
   std::vector<const char*> argv;
   argv.push_back("amber");
-  argv.push_back("-d");
 
-  for (auto s : argv_string)
-    argv.push_back(s.c_str());
+  for (size_t i = 0; i < argv_string.size(); i++)
```

Comment:
Nit but could you use a `for(const std::string& arg : argv_string)` loop instead?

Reply:
Sure